### DISCUSSION TITLE
Blockchain-based storage backend for auditable workflows

### DISCUSSION THREAD

**Discussion by wackazong** [mlflow/mlflow#11246] [Ideas]
I am completely new to mlflow, coming here from the blockchain space. I heard from a number of people using mlflow that it would make a lot of sense to use a blockchain-based storage backend for mlflow in order to make the whole process auditable and transparent. 

What are your thoughts on this? Is this a thing? Would it be a much welcomed addition to mlflow? What would be the potential use cases?

Being the co-founder of a blockchain-based startup for enterprise business processes (https://ibexus.io), I could see some potential collaborations here. But I would need to know more about what people actually need. Therefore, if you can spare a minute to describe some potential applications, I would very much appreciate it.

**Comment by kempsterrrr** [comment]
Hey @wackazong did you go anywhere with this?

I've do see value in using decentralised solutions, primarily for increasing independent verifiability and lowering some of the trust assumption and have built a pugin that anchors ML FLow life-cycle events as signed commitments to Arweave - https://github.com/mlflow/mlflow/discussions/23355

Love to know if you did anything here and any feedback :)

**Reply by wackazong** [reply]
Unfortunately not, as this was not a priority. But #23355 looks promising!
